api/edge/v1alpha1/types.go

```diff
@@ -171,6 +173,133 @@ type PlacementList struct {
+type PlacementDecisionSpec struct {
+	// `Workload` is a collection of namespaced-scoped resources and a collection of cluster-scoped resources to be propagated to destination clusters.
+	Workload PlacementDecisionSpecResources `json:"workload,omitempty"`
+
+	// `destinations` is a list of cluster-identifiers that the resources should be propagated to.
+	Destinations []Destination `json:"destinations,omitempty"`
+}
+
+// PlacementDecisionSpecResources explicitly defines the resources to be down-synced.
+// The ClusterScope list defines the cluster-scope resources, while NamespacedObjects packs individual resources
+// identifiable by namespace & name.
+type PlacementDecisionSpecResources struct {
+	// `clusterScope` holds a list of individual cluster-scoped objects
+	// to downsync, organized by resource.
+	// Remember that a "resource" is a kind/type/sort of objects,
+	// not an individual object.
+	// +optional
+	ClusterScope []ClusterScopeDownsyncResource `json:"clusterScope,omitempty"`
+
+	// `NamespaceScope` matches if and only if at least one member matches.
+	// +optional
+	NamespaceScope []NamespaceScopeDownsyncObjects `json:"namespaceScope,omitempty"`
+}
+
+// NamespaceScopeDownsyncObjects matches some objects of one particular namespaced resource.
+type NamespaceScopeDownsyncObjects struct {
+	// GroupResource holds the API group and resource name.
+	metav1.GroupResource `json:",inline"`
+
+	// `apiVeresion` holds just the version, not the group too.
+	// This is the version to use both upstream and downstream.
+	APIVersion string `json:"apiVersion"`
+
+	// `objectsByNamespace` matches by namespace and name.
+	// An object matches the list if and only if the object matches at least one member of the list.
+	// Thus, no object matches the empty list.
+	// +optional
+	ObjectsByNamespace []NamespaceAndNames `json:"objectsByNamespace,omitempty"`
+}
+
+// NamespaceAndNames identifies some objects of an implied resource that is namespaced.
+// The objects are all in the same namespace.
+type NamespaceAndNames struct {
+	// `namespace` identifies the namespace
+	Namespace string `json:"namespace"`
+
+	// `names` holds the names of the objects that match.
+	// Empty list means none of them.
+	// +optional
+	Names []string `json:"names,omitempty"`
+}
+
+type ClusterScopeDownsyncResource struct {
+	// GroupResource holds the API group and resource name.
+	metav1.GroupResource `json:",inline"`
+
+	// `apiVeresion` holds just the version, not the group too.
+	// This is the version to use both upstream and downstream.
+	APIVersion string `json:"apiVersion"`
+
+	// `objects` holds the names of the objects of this kind to downsync.
+	// Empty list means none of them.
+	// +optional
+	Objects []string `json:"objects,omitempty"`
+}
+
+// Destination wraps the identifiers required to uniquely identify a destination cluster.
+type Destination struct {
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
So I was expecting something like a cluster name?

Reply:
changed it to clusterId

Reply:
yes